src/Bicep.Cli/TrimmerRoots.xml

```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<linker>
```

Comment:
>linker [](http://example.com/codeflow?start=1&length=6)

The trimmer behavior has changed from .net 5 to .net 6 and there is no method to restore .net 5 behavior. In .net 5 the trimmer would only trim assemblies if they were not used. By default in .net 6, the trimming happens at member level. (Unusued members get removed as determined by the trimmer.)

This sounds as risky as it sounds. Any dependency on members that can't be discovered via static analysis (for example reflection) can potentially break with this. The .net team has helpfully provided an analyzer that flags code in our app and inside dependencies that can potentially break the trimmer. The most common offenders are our Newtonsoft and System.Text.Json usages as well some reflection we do in the CLI and linter to discover classes drived from a common base class or interface. Overall, we end up with numerous warnings that will take some time to get through. We will need to clear them up because the long term plan with .net 7 or .net 8 appears to be to only allow member-level trimming. (This is the `link` trimmer mode.) If we don't do it, we'll have issues upgrading to .net 7 and 8 in the future.

To unblock progress I switched the trimmer to use `copyused` mode, which approximates .net 5 behavior but still has some differences. (Keeps the entire assembly if it's found to be used.) Even with that setting, E2E and live tests had issues with missing types. By debugging, I was able to identify the missing assemblies and "rooted" them here via this file. Eventually I ended up with enough assemblies that all of our e2e tests pass. Given that the scope of the "rooting" is an assembly, it should be safer than the default member-level trimming and require less maintenance. 

Going forward, we will need to do the following:
- continue investments in e2e tests to ensure we catch any regressions caused by assembly over-trimming
- replace all System.Text.Json reflection-based usages with a source generator (available in a nuget)
- come up with sufficient trimmer annotations for Newtonsoft and other dependencies 
- replace in-app reflection with a custom source generator
- ban all reflection in code used by the CLI and WASM app. (Test code can still use reflection, for example.)

The upside of all this is the trimmed version of the Bicep CLI is down to 35 MB from previous 46MB size and we can probably get it even lower.